lta/bundler.py

```diff
@@ -176,38 +177,43 @@ def _check_mysql(self) -> bool:
-        self.logger.info(f"Computing LTA checksums for bundle: '{final_bundle_path}'")
-        checksum = lta_checksums(final_bundle_path)
-        self.logger.info(f"Bundle '{final_bundle_path}' has adler32 checksum '{checksum['adler32']}'")
-        self.logger.info(f"Bundle '{final_bundle_path}' has SHA512 checksum '{checksum['sha512']}'")
-        # 3.3. Clean up generated JSON metadata file
+        # 3. Clean up generated JSON metadata file
         self.logger.info(f"Deleting bundle metadata file: '{metadata_file_path}'")
         os.remove(metadata_file_path)
         self.logger.info(f"Bundle metadata '{metadata_file_path}' was deleted.")
-        # 4. Update the Bundle in the LTA DB
+        # 4. Compute the size of the bundle
+        bundle_size = os.path.getsize(bundle_file_path)
+        self.logger.info(f"Archive bundle has size {bundle_size} bytes")
+        # 5. Compute the LTA checksums for the bundle
+        self.logger.info(f"Computing LTA checksums for bundle: '{bundle_file_path}'")
+        checksum = lta_checksums(bundle_file_path)
+        self.logger.info(f"Bundle '{bundle_file_path}' has adler32 checksum '{checksum['adler32']}'")
+        self.logger.info(f"Bundle '{bundle_file_path}' has SHA512 checksum '{checksum['sha512']}'")
+        # 6. Determine the final destination path of the bundle
+        final_bundle_path = bundle_file_path
+        if self.outbox_path != self.workbox_path:
+            final_bundle_path = os.path.join(self.outbox_path, f"{bundle_id}.zip")
+        self.logger.info(f"Finished archive bundle will be located at: '{final_bundle_path}'")
+        # 7. Update the bundle record we have with all the information we collected
         bundle["status"] = "created"
         bundle["update_timestamp"] = now()
         bundle["bundle_path"] = final_bundle_path
         bundle["size"] = bundle_size
         bundle["checksum"] = checksum
         bundle["verified"] = False
         bundle["claimed"] = False
-        self.logger.info(f"PATCH /Bundles/{bundle_id} - '{bundle}'")
-        await lta_rc.request('PATCH', f'/Bundles/{bundle_id}', bundle)
+        # 7. Add a row to the JADE-LTA database with the bundle information
         # NOTE: Remove two lines below when JADE LTA is retired
         self.logger.info(f"Adding row to MySQL Database: {self.user}@{self.host}/{self.db}")
         self._insert_jade_row(bundle)
+        # 8. Move the bundle from the work box to the outbox
+        if final_bundle_path != bundle_file_path:
+            self.logger.info(f"Moving bundle from '{bundle_file_path}' to '{final_bundle_path}'")
+            shutil.move(bundle_file_path, final_bundle_path)
+        self.logger.info(f"Finished archive bundle now located at: '{final_bundle_path}'")
+        # 9. Update the Bundle record in the LTA DB
+        self.logger.info(f"PATCH /Bundles/{bundle_id} - '{bundle}'")
+        await lta_rc.request('PATCH', f'/Bundles/{bundle_id}', bundle)
```

Comment:
So for JADE LTA it's required that the DB row is created before the file is moved?

Reply:
Not strictly so; it's possible to adjust the glue.

However, the JADE pattern is very much file handoff is ownership handoff. Doing work after moving the file to the outbox invites race conditions.

In LTA, we do it the opposite way. We're allowed to move the file around as much as we want, up until we do the handoff in the REST server. Doing work after updating the REST server would invite the race condition.

Reply:
Ah, OK.